src/components/Survey.js

```diff
@@ -1,65 +1,110 @@
+  const {handleSubmit} = props 
+  const navigation = useNavigation();
+  const dispatch = useDispatch() 
+ 
+  
 
-const handleSubmit = (valueP) => {
-  Alert.alert("Mis valores props handleSubmit son ",valueP)
+  const submit = async (values) => {
+    
+    if(!checkFields(userName,cell,comment)){
+      Alert.alert("Tiene que llenar todos los campos")
+      return 0
+    
+    }else{
+
+    const keyResponse = { input_values: { values } };
+    const response = await axios.post('https://private-e75208-formresponse.apiary-mock.com/form_response', keyResponse);
+    if(response != null){
+      Alert.alert("¡Gracias por el mensaje!")
+    }
+  }
 }
 
+  const handleOnChange = (user) => {
+    setUserName(user)
+  }
+
+  const savePhone = (user) => {
+    setCell(user)
+  }
+  
+  const saveComment = (user) => {
+    setComment(user)
+  }
+
+  
+  const destroyForm = () => {
+    dispatch(saveUser(userName))         
+    destroy('survey')
+    navigation.navigate({name:'Home'})     
+  }
+
+ /*  console.log(userName, cell, comment) */
+
+       
+
     return (
     <>
+       <SafeAreaView style={{ flex: 1, backgroundColor: '#19191b' }}>
+        <ScrollView>
           <View style={{flex: 1, flexDirection:'column',margin:40,justifyContent:'flex-start'}}>
             <Text style={styles.titleForm}>Formulario</Text>
             
-            <Field label='Nombre' component={renderField} name={"nombre"} />
-            <Field label='Telefono' component={renderField} name={"telefono"} />
-            <Field label='Comentario' component={renderField} name={"comentarios"} />
+            <Field label='Usuario' component={RenderField} name={"userName"} validate={[required, checkUser]} onChange={handleOnChange}/>
+            <Field label='Telefono' component={RenderField} name={"telefono"} validate={[required, checkTel, telLength]} onChange={savePhone}/>
+            <Field label='Comentario' component={RenderField} name={"comentarios"} validate={required} onChange={saveComment}/>
 
        <View style={{flexDirection:'row', margin:10, alignItems:'center'}}>
        
-          <TouchableOpacity style={styles.cardButton}>
+          <TouchableOpacity style={styles.cardButton} onPress={handleSubmit(submit)}>
             <Text style={styles.buttonSubmit}>Submit</Text>
           </TouchableOpacity>
 
-          <TouchableOpacity onPress={() => console.log("Clikeaste Cancel")} style={styles.cardButton}>
+          <TouchableOpacity onPress={destroyForm} style={styles.cardButton}>
             <Text style={styles.buttonCancel}>Cancel</Text>
           </TouchableOpacity>
        
           </View>
       </View>
+      </ScrollView>
+      </SafeAreaView>
     </>
     )
   }
   
   Survey = reduxForm({
-    form: 'survey'
+    form: 'survey',
+    enableReinitialize: true,
   })(Survey) 
 
+   Survey = connect(
+    state => ({
+      initialValues: state.opRed,
+      storeUserName: state.opRed.userName,
+  }))(Survey); 
```

Comment:
Be consistent with using hooks or hocs. If you are using `useSelector` and `useDispatch` then use it everywhere. Do not mix them.

Reply:
This point has not been entirely clear to me I would like a detailed explanation to understand how I can not mix the HOC